src/tests/test_log_client.py

```diff
@@ -0,0 +1,226 @@
+        t = 0.0
+        pid = 9999999
+        name = self.log_client.name
+        args = [e_errors.MISC + 1, "an log message", "info"]
+        self.log_client.log_func(t, pid, name, args)
+        unique_id = True
+        p1 = {
+            'message': '9999999 %s M LOG_CLIENT  6 an log message' % self.user,
+            'work': 'log_message'}
+        p2 = ('131.225.214.78', 7504)
+        self.sent_msg.assert_called_with(p1, p2, unique_id=True)
+
+    def test_set_get_logpriority(self):
+        # NB the log server reacts to the log priority
+        # this needs to be tested with the log server running
+        # or in the log server test
+        log_prio = 99999
+        self.assertNotEqual(self.log_client.get_logpriority(), log_prio)
+        self.log_client.set_logpriority(log_prio)
+        self.assertEqual(self.log_client.get_logpriority(), log_prio)
+
+    def test_get_logfile_name(self):
+        self.sent_msg.reset_mock()
+        self.log_client.get_logfile_name()
+        self.sent_msg.assert_called_with(
+            {'work': 'get_logfile_name'}, ('131.225.214.78', 7504), 0, 0)
+
+    def test_get_logfiles(self):
+        self.sent_msg.reset_mock()
+        self.log_client.get_logfiles('today')
+        self.sent_msg.assert_called_with(
+            {'work': 'get_logfiles', 'period': 'today'}, ('131.225.214.78', 7504), 0, 0)
+
+    def test_get_last_logfile_name(self):
+        self.sent_msg.reset_mock()
+        self.log_client.get_last_logfile_name()
+        self.sent_msg.assert_called_with(
+            {'work': 'get_last_logfile_name'}, ('131.225.214.78', 7504), 0, 0)
+
+
+class TestTCPLoggerClient(unittest.TestCase):
+    def setUp(self):
+        def tmp_dir_side_effect(*args, **kwargs):
+            return "/tmp"
+        enstore_functions.get_enstore_tmp_dir = mock.MagicMock()
+        enstore_functions.get_enstore_tmp_dir.side_effect = tmp_dir_side_effect
+        threading.Thread = mock.MagicMock()
+        self.sent_msg = mock.MagicMock()
+        self.user = pwd.getpwuid(os.getuid())[0]
+        csc = mock_csc.csc()
+        self.log_client = log_client.TCPLoggerClient(csc)
+
+    def test___init__(self):
+        self.assertTrue(
+            isinstance(
+                self.log_client,
+                log_client.TCPLoggerClient))
+
+    def test_log_func(self):
+        self.sent_msg.reset_mock()
+        t = 0.0
+        pid = 9999999
+        name = self.log_client.name
+        args = [e_errors.MISC + 1, "an log message", "info"]
+        self.log_client.message_buffer.put_nowait = self.sent_msg
+        self.log_client.log_func(t, pid, name, args)
+        p1 = {
+            'message': '9999999 %s M LOG_CLIENT  6 an log message' % self.user,
+            'work': 'log_message',
+            'sender': mock.ANY}
+        self.sent_msg.assert_called_with(p1)
+
+    def test_stop(self):
+        self.log_client.stop()
+        self.assertFalse(self.log_client.run)
+
+    def test_connect(self):
```

Comment:
I think we should check the error cases here as well, I know mocking this socket stuff sucks but let's test:
socket.socket.connect.side_effect => socket.error EISCONN: success
socket.socket.connect.side_effect => socket.error ECONNREFUSED: throws ECONNREFUSED
select.select.return_value => (None, None, None): throws ETIMEDOUT
socket.getsockopt.return_value => 1: throws socket.error(1, os.strerror(1))

Reply:
This turned out to be worth testing.  I tested EACCES (permission denied) instead of ECONNREFUSED but the pathways were the same.   Testing the select.select timeout as coded returned "TypeError: exceptions must be old-style classes or derived from BaseException, not int" so that was fixed that in log_client.py

Reply:
Thanks!